datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -164,3 +166,177 @@ pub fn concat_elements_utf8view(
     }
     Ok(result.finish())
 }
```

Comment:
#11570 talks about implementing a new kernel for the divide-if-gt-zero operation and I took a stab at it but realized it was pretty involved. Or maybe I'm not reading the right bits. It looks to me like the existing kernels for binary operations in `arrow-rs` don't support `Option` as a return. I've left this here to illustrate the kind of things that I *think* have to be added to make a whole new kernel for this. Though `arrow-rs` is a big project and I'm probably missing something.